unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Passport/Passport/PassportNavigation/PassportNavigationComponentController.cs

```diff
@@ -129,17 +151,16 @@ async UniTaskVoid RequestOwnedWearablesAsync(CancellationToken ct)
             {
                 try
                 {
+                    view.SetViewAllButtonActive(PassportSection.Wearables, false);
                     var wearables = await wearablesCatalogService.RequestOwnedWearablesAsync(
                         userProfile.userId,
                         1,
-                        int.MaxValue,
+                        MAX_NFT_COUNT + 1,
                         true,
                         ct);
 
-                    IGrouping<string, WearableItem>[] wearableItems = wearables.GroupBy(i => i.id).ToArray();
-                    string[] wearableIds = wearableItems.Select(g => g.First().id).Take(MAX_NFT_COUNT).ToArray();
-                    userProfile.SetInventory(wearableIds);
-                    loadedWearables.AddRange(wearableIds);
+                    view.SetViewAllButtonActive(PassportSection.Wearables, wearables.totalAmount > MAX_NFT_COUNT);
+                    IGrouping<string, WearableItem>[] wearableItems = wearables.wearables.GroupBy(i => i.id).ToArray();
```

Comment:
I propose to take a look at this expression and the following `ToArray` as they allocate too much.

Reply:
That's true... I have been able to avoid it. Fixed in: https://github.com/decentraland/unity-renderer/pull/4316/commits/605c13d5f0cb66aa5621b9865faa157a1e790121